Public/Src/Cache/ContentStore/Library/FileSystem/TrackingFileStream.cs

```diff
@@ -0,0 +1,163 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Security.AccessControl;
+using System.Threading;
+using JetBrains.Annotations;
+using Microsoft.Win32.SafeHandles;
+
+namespace BuildXL.Cache.ContentStore.FileSystem
+{
+    /// <summary>
+    /// Special tracking file stream that fails with more readable error message if unhandled error occurs in the finalizer of the instance.
+    /// </summary>
+    internal class TrackingFileStream : FileStream

```

Comment:
Is it worth considering a more generic IDisposableWarpper?

`class LeakDetection<T> : T, IDisposable`